runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineDebugOptions.java

```diff
@@ -216,6 +216,25 @@ public Dataflow create(PipelineOptions options) {
 
   void setDumpHeapOnOOM(boolean dumpHeapBeforeExit);
 
+  /**
+   * If true, save a JFR profile when GC thrashing is first detected. The profile will run for the
+   * amount of time set by --jfrRecordingDurationSec, or 60 seconds by default.
+   *
+   * <p>Note, JFR profiles are only supported on java 9 and up.
+   */
+  @Description(
+      "If true, save a JFR profile before killing a thread or process "
+          + "which is GC thrashing or out of memory.  Only available on java 9 or up")
+  boolean getRecordJfrOnGcThrashing();
```

Comment:
This feature is nice, can we add it to sdks/java/harness as well. 

See https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/options/SdkHarnessOptions.java and https://github.com/apache/beam/blob/master/sdks/java/harness/src/main/java/org/apache/beam/fn/harness/status/MemoryMonitor.java

Reply:
heh yeah I mentioned that in my comment, I'll do it in another review once this is accepted/landed.  I didn't want to apply the same changes twice. :P

Reply:
Sorry, should have read that. :-)

Reply:
its cool :D